mack/__init__.py

```diff
@@ -315,12 +320,27 @@ def delta_file_sizes(delta_table: DeltaTable):
 
 def humanize_bytes(n: int) -> str:
     for prefix, k in (
-        ("PB", 1e15),
-        ("TB", 1e12),
-        ("GB", 1e9),
-        ("MB", 1e6),
-        ("kB", 1e3),
+            ("PB", 1e15),
+            ("TB", 1e12),
+            ("GB", 1e9),
+            ("MB", 1e6),
+            ("kB", 1e3),
     ):
         if n >= k * 0.9:
             return f"{n / k:.2f} {prefix}"
     return f"{n} B"
+
+
+def find_composite_key(df, exclude_cols: List[str] = None):
+    if exclude_cols is None:
+        exclude_cols = []
+    df_col_excluded = df.drop(*exclude_cols)
+    total_row_count = df_col_excluded.distinct().count()
+    for n in range(1, len(df_col_excluded.columns) + 1):
+        for c in combinations(df_col_excluded.columns, n):
+            if df_col_excluded.select(*c).distinct().count() == total_row_count:
+                return list(df_col_excluded.select(*c).columns)
+
+
+def get_md5(df, list_of_columns):
```

Comment:
Let's call this `with_md5_cols` or something like that.  `get_whatever` methods have the [Java getter methods](https://www.w3schools.com/java/java_encapsulation.asp) connotation, which is a lot different than adding a column to a DataFrame.